### PR TITLE
[feature] Add aws-redis-replication-group

### PR DESCRIPTION
**Pull request by davidrissato**
### Summary
This module is similar to [aws-redis-node](https://github.com/chanzuckerberg/cztack/tree/main/aws-redis-node), however it creates a Elasticache Redis cluster using a replication group.
The motivation to create this module is because encryption options (in-transit and at rest) are only available when you create your cluster using a replication group.

### Test Plan
- Manual tests (already applied to IDseq)
- Unit tests

### References
(Optional) Additional links to provide more context.

### CODE REVIEW

File: aws-redis-replication-group/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+output "primary_endpoint_address" {
```
Comment (cziie) [inline aws-redis-replication-group/outputs.tf:1]: 📝 **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>`primary_endpoint_address` output has no description

File: aws-redis-replication-group/outputs.tf [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  value = aws_elasticache_replication_group.default.primary_endpoint_address
+}
+
+output "configuration_endpoint_address" {
```
Comment (cziie) [inline aws-redis-replication-group/outputs.tf:5]: 📝 **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>`configuration_endpoint_address` output has no description

File: aws-redis-replication-group/outputs.tf [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  value = aws_elasticache_replication_group.default.configuration_endpoint_address
+}
+
+output "port" {
```
Comment (cziie) [inline aws-redis-replication-group/outputs.tf:9]: 📝 **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>`port` output has no description

**Review by ryanking — CHANGES_REQUESTED** [review]
This module needs to be added to [the list in the ci.yml](https://github.com/chanzuckerberg/cztack/blob/davidrc/redis-replication-group/.github/workflows/ci.yml#L74) in order to run the tests in CI.

**Review by ryanking — APPROVED** [review]

**Pull request merged**